src/odemis/gui/comp/canvas.py

```diff
@@ -952,6 +952,9 @@ def _draw_merged_images(self, ctx, interpolate_data=False):
                 if isinstance(im, tuple):
                     first_tile = im[0][0]
                     md = first_tile.metadata
+                    tiles_merged_shape = util.img.getTilesSize(im)
```

Comment:
I'd prefer to have this code move to miccanvas, in _convert_streams_to_images(), where there used to be `getBoundingBox()`. This means, that canvas doesn't indirectly depend on having MD_ROTATION, MD_POS, MD_SHEAR on the first tile. In addition, another (small) advantage is that'll be computed only once for all the times the same set of tiles is displayed.
Then, to store the metadata (in this class), just put the metadata on the first tile, as a convention.